packages/forgetti/runtime/index.ts

```diff
@@ -1,19 +1,51 @@
+function isEqual(a: unknown, b: unknown): boolean {
+  // eslint-disable-next-line no-self-compare
+  return a === b || (a !== a && b !== b);
+}
+
 export function $$equals(cache: unknown[], index: number, b: unknown): boolean {
-  if (index in cache) {
-    const a = cache[index];
-    // eslint-disable-next-line no-self-compare
-    return a === b || (a !== a && b !== b);
-  }
-  return false;
+  return index in cache && isEqual(cache[index], b);
 }
 
-export type MemoHook = <T>(callback: () => T, dependencies: unknown[]) => T;
+export interface Ref<T> {
+  current: T;
+}
+export type RefHook = <T>(callback: T) => Ref<T>;
 
-export function $$cache(hook: MemoHook, size: number): unknown[] {
-  return hook(() => new Array<unknown>(size), []);
+export function $$cache(hook: RefHook, size: number): unknown[] {
+  const ref = hook<unknown[] | undefined>(undefined);
+  if (!ref.current) {
+    ref.current = new Array(size);
+  }
+  return ref.current;
 }
 
 export function $$branch(parent: unknown[], index: number, size: number): unknown[] {
   parent[index] ||= new Array(size);
   return parent[index] as unknown[];
 }
+
+export interface MemoProps {
+  value: unknown;
+}
```

Comment:
Since:

- Object property `value` can not be minified
- We are basically caching every JSX

IMHO we should use a shorter prop name for this. I propose `r` as `r` in `render prop`.

Reply:
done, I just realized it cannot be mangled